### PR TITLE
tests for 'optional catch binding' proposal, stage 3

### PR DESCRIPTION
**Pull request by dnalborczyk**
**optional catch binding**, stage 3

supported by Safari TP

https://github.com/tc39/proposals
https://github.com/tc39/proposal-optional-catch-binding
https://tc39.github.io/proposal-optional-catch-binding/

### CODE REVIEW

File: data-esnext.js [hunk 1]
```diff
@@ -2653,6 +2653,64 @@ exports.tests = [
...
+  subtests: [
+    {
+      name: 'basic',
+      exec: function(){/*
+        try {
+          throw new Error();
+        }
+        catch {
+          return true;
+        }
```
Comment (ljharb) [inline data-esnext.js:2671]: should this have an explicit `return false` at the end?

File: data-esnext.js [hunk 2]
```diff
@@ -2653,6 +2653,64 @@ exports.tests = [
...
+            yield;
+          }
+          catch {
+            return true;
+          }
+        }
+
+        var it = foo();
+        it.next();
+        return it.throw().value;
```
Comment (ljharb) [inline data-esnext.js]: this seems like it requires proper implementation of `.throw()` - could we test this with `yield; throw new Error();` instead?

File: data-esnext.js [hunk 3]
```diff
@@ -2653,6 +2653,64 @@ exports.tests = [
...
+      exec: function(){/*
+        try {
+          throw new Error();
+        }
+        catch {
+          return true;
+        }
+      */},
+      res: {
+        safaritp: true
```
Comment (chicoxyzzy) [inline data-esnext.js]: WebKit Nightly also supports these features

File: data-esnext.js (hunk 1 above)
Comment (dnalborczyk) [reply data-esnext.js:2671]: I assumed an implicit "return undefined" (false) -- as with the other 2 tests -- but I don't have a strong opinion about it, hence I added it.

**Review by chicoxyzzy — APPROVED** [review]

**Comment by chicoxyzzy** [comment]
I just merged another PR and it introduced conflicts. Sorry for that :(

**Comment by dnalborczyk** [comment]
no problem. Just pulled upstream, and rebuild the html.

**Review by ljharb — APPROVED** [review]

**Comment by chicoxyzzy** [comment]
@dnalborczyk thank you for your contribution!

**Pull request merged**